x-pack/plugins/maps/public/classes/sources/es_search_source/es_search_source.tsx

```diff
@@ -542,17 +619,24 @@ export class ESSearchSource extends AbstractESSource {
     };
   }
 
-  getSyncMeta() {
+  getSyncMeta(): VectorSourceSyncMeta | null {
     return {
-      sortField: this._descriptor.sortField,
-      sortOrder: this._descriptor.sortOrder,
+      sortField: typeof this._descriptor.sortField === 'string' ? this._descriptor.sortField : '',
```

Comment:
Logic is not needed. `descriptor.sortField` is defaulted to empty string in constructor